pygmt/datasets/samples.py

```diff
@@ -1,13 +1,67 @@
 """
 Functions to load sample data.
 """
+import warnings
+
 import pandas as pd
+from pygmt.exceptions import GMTInvalidInput
 from pygmt.src import which
 
 
+def list_sample_dataframes():
+    """
+    Report tabular datasets available for tests and documentation examples.
+
+    Returns
+    -------
+    output : dict
+        Names and short descriptions of available sample dataframes.
+    """
+    names = {
+        "japan_quakes": "Table of earthquakes around Japan from NOAA NGDC database",
+        "ocean_ridge_points": "Table of ocean ridge points for the entire world",
+    }
+    return names
+
+
+def load_sample_dataframe(name):
+    """
+    Load an example dataset from the GMT server.
+
+    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
+    first time you invoke this function. Afterwards, it will load the data from
+    the cache. So you'll need an internet connection the first time around.
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset to load.
+
+    Returns
+    -------
+    output : pandas.DataFrame
+        Tabular dataset.
+    """
+    names = list_sample_dataframes()
+    if name not in names:
+        raise GMTInvalidInput(f"Invalid dataset name '{name}'")
+
+    if name == "japan_quakes":
+        data = load_tut_quakes()
+
+    if name == "ocean_ridge_points":
+        data = load_ridge()
+
+    return data
+
+
 def load_japan_quakes():
     """
-    Load a table of earthquakes around Japan as a pandas.DataFrame.
+    Load a table of earthquakes around Japan as a pandas.DataFrame (Deprecated)
```

Comment:
Maybe put (Deprecated) at the beginning of the sentence so that it stands out in the API docs list?
```suggestion
    (Deprecated) Load a table of earthquakes around Japan as a pandas.DataFrame.
```